libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranteeItem.tsx

```diff
@@ -170,13 +171,30 @@ const GranteeGroupItem: React.FC<IGranteeGroupItemProps> = (props) => {
  * @internal
  */
 export const GranteeItemComponent: React.FC<IGranteeItemProps> = (props) => {
-    const { grantee, mode, areGranularPermissionsSupported, onDelete, onChange } = props;
+    const { grantee, mode, areGranularPermissionsSupported, currentUserPermissions, onDelete, onChange } =
+        props;
 
     if (areGranularPermissionsSupported) {
```

Comment:
Is this check necessary ? It could simplify visitor pattern next. All types has ist type guards.

Reply:
I made the grantee types discriminating, the check is gone now.